apps/advisor/pkg/app/checks/datasourcecheck/prom_dep_auth_check_step.go

```diff
@@ -0,0 +1,162 @@
+package datasourcecheck
+
+import (
+	"context"
+	"fmt"
+	sysruntime "runtime"
+
+	"github.com/grafana/grafana-app-sdk/logging"
+	advisor "github.com/grafana/grafana/apps/advisor/pkg/apis/advisor/v0alpha1"
+	"github.com/grafana/grafana/apps/advisor/pkg/app/checks"
+	"github.com/grafana/grafana/pkg/plugins/repo"
+	"github.com/grafana/grafana/pkg/services/datasources"
+)
+
+type promDepAuthStep struct {
+	PluginRepo     repo.Service
+	GrafanaVersion string
+}
+
+func (s *promDepAuthStep) Title() string {
+	return "Prometheus deprecated authentication check"
+}
+
+func (s *promDepAuthStep) Description() string {
+	return "Check if Prometheus data sources are using deprecated authentication methods (Azure auth and SigV4)"
+}
+
+func (s *promDepAuthStep) Resolution() string {
+	return "Enable the feature toggle for 'prometheusTypeMigration'. If this feature toggle is already enabled, install 'Azure Monitor Managed Service for Prometheus' and/or 'Amazon Managed Service for Prometheus' data sources."
+}
+
+func (s *promDepAuthStep) ID() string {
+	return PromDepAuthStepID
+}
+
+func (s *promDepAuthStep) Run(ctx context.Context, log logging.Logger, obj *advisor.CheckSpec, item any) ([]advisor.CheckReportFailure, error) {
+	dataSource, ok := item.(*datasources.DataSource)
+	if !ok {
+		return nil, fmt.Errorf("invalid item type %T", item)
+	}
+	if dataSource.Type != datasources.DS_PROMETHEUS {
+		return nil, nil
+	}
+	if dataSource.JsonData == nil {
+		return nil, nil
+	}
+
+	awsAuthLinks, err := s.checkUsingAWSAuth(ctx, dataSource)
+	if err != nil {
+		return nil, err
+	}
+	azureAuthLinks, err := s.checkUsingAzureAuth(ctx, dataSource)
+	if err != nil {
+		return nil, err
+	}
+
+	errorLinks := append(awsAuthLinks, azureAuthLinks...)
+
+	if len(errorLinks) == 0 {
+		return nil, nil
+	}
+
+	return []advisor.CheckReportFailure{checks.NewCheckReportFailureWithMoreInfo(
+		advisor.CheckReportFailureSeverityHigh,
+		s.ID(),
+		dataSource.Name,
+		dataSource.UID,
+		errorLinks,
+		fmt.Sprintf("Plugin: %s", dataSource.Type),
+	)}, nil
+}
+
+func (s *promDepAuthStep) checkUsingAWSAuth(ctx context.Context, dataSource *datasources.DataSource) ([]advisor.CheckErrorLink, error) {
+	var errorLinks []advisor.CheckErrorLink
+	if sigV4Auth, found := dataSource.JsonData.CheckGet("sigV4Auth"); found {
+		if enabled, err := sigV4Auth.Bool(); err != nil || !enabled {
+			return nil, err
```

Comment:
I'd avoid the error here (for the same reason explained above, to avoid erroring the whole check):


```suggestion
			// Disabled or not a valid boolean
			return nil, nil
```